source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1054 @@
+                        slope = bottom_flux/kdd - std::tan(slopep[index_bot+nx]*boost::math::double_constants::pi/180);
+                      else if (j==(nx-1))
+                        slope = bottom_flux/kdd - std::tan(slopep[index_bot+nx-2]*boost::math::double_constants::pi/180);
+                      else
+                        slope = bottom_flux/kdd - std::tan(slopep[index_bot+nx-1]*boost::math::double_constants::pi/180);
+
+                      h[index_bot-1] = h[index_bot-1] + slope*2*dx;
+                    }
+
+                  if (bottom == 0 && top == 0)
+                    {
+                      double side = index_bot;
+                      int jj = nx;
+
+                      if (vy[index_bot+nx-1] > 0 && vy[index_top-nx-1] >= 0)
+                        {
+                          side = index_top;
+                          jj = -nx;
+                        }
+                      else if (vy[index_bot+nx-1] <= 0 && vy[index_top-nx-1] < 0)
+                        {
+                          side = index_bot;
+                          jj = nx;
+                        }
+                      else
+                        continue;
+
+                      vz[index_bot-1] = vz[side+jj-1];
+                      vz[index_top-1] = vz[side+jj-1];
+
+                      vy[index_bot-1] = vy[side+jj-1];
+                      vy[index_top-1] =  vy[side+jj-1];
+
+                      vx[index_bot-1] = vx[side+jj-1];
+                      vx[index_top-1] =  vx[side+jj-1];
+
+                      h[index_bot-1] = h[side+jj-1];
+                      h[index_top-1] = h[side+jj-1];
+                    }
+                }
+
+
+              //Find a fastscape timestep that is below our maximum timestep.
+              double f_dt = a_dt/steps;
+              while (f_dt>max_timestep)
+                {
+                  steps=steps*2;
+                  f_dt = a_dt/steps;
+                }
+
+              //Set time step
+              fastscape_set_dt_(&f_dt);
+
+              //Set velocity components and h.
+              fastscape_set_u_(vz.get());
+              fastscape_set_v_(vx.get(), vy.get());
+              fastscape_set_h_(h.get());
+
+              //Initialize first time step, and update steps.
+              int visualization_step = istep+restart_step;
+              steps = istep+steps;
+
+              this->get_pcout() <<"   Calling FastScape... "<<(steps-istep)<<" timesteps of "<<f_dt<<" years."<<std::endl;
+              {
+                auto t_start = std::chrono::high_resolution_clock::now();
+
+                /*
+                 * If we use stratigraphy it'll handle visualization and not the normal function.
+                 * TODO: The frequency in this needs to be the same as the total timesteps fastscape will
+                 * run for, need to figure out how to work this in better.
+                 */
+                if (use_strat && current_timestep == 1)
```

Comment:
I think stratigraphy will need to be revisited, but I believe the reasoning behind this was that you only need to call strati at the start and then it will handle all the visualization aspects. If you don't call it, then the VTK function needs to be called every time you want to make a visualization.